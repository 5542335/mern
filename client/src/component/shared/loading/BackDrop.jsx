import React, { useCallback, useState } from 'react';
import Backdrop from '@material-ui/core/Backdrop';
import CircularProgress from '@material-ui/core/CircularProgress';

import style from './backDrop.module.css';

export const SimpleBackdrop = () => {
  const [open, setOpen] = useState(true);
  const handleClose = useCallback(() => {
    setOpen(false);
  }, []);

  return (
    <Backdrop className={style.backdrop} open={open} onClick={handleClose}>
      <CircularProgress color="secondary" />
    </Backdrop>
  );
};
