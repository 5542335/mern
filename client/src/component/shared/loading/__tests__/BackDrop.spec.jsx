import React from 'react';
import { shallow } from 'enzyme';

import { SimpleBackdrop } from '../BackDrop';

describe('Component: BackDrop', () => {
  describe('Test default state', () => {
    it('Should render component', () => {
      const component = shallow(<SimpleBackdrop />);

      expect(component).toMatchSnapshot();
    });

    it('Should render component with default open state === true', () => {
      const component = shallow(<SimpleBackdrop />);

      expect(component.props().open).toStrictEqual(true);
    });
  });

  describe('Test click on component', () => {
    it('Should set open to false', () => {
      const component = shallow(<SimpleBackdrop />);

      component.simulate('click');

      expect(component.props().open).toStrictEqual(false);
    });
  });
});
