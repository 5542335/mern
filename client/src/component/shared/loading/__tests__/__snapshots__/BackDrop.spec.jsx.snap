// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BackDrop Test default state Should render component 1`] = `
<WithStyles(ForwardRef(Backdrop))
  className="backdrop"
  onClick={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(CircularProgress))
    color="secondary"
  />
</WithStyles(ForwardRef(Backdrop))>
`;
